class_configs/shd_class_config.lua

```diff
@@ -641,82 +651,156 @@ local _ClassConfig = {
 
             return res
         end,
+        --Handles all AE taunt checks rather than repetetive hardcoded conditions in the UI loop
+        AETauntCheck = function(self)
+            --check to see if we are the tank in the first place just in case something CRAAAAAZY happens
+            if not RGMercUtils.IsTanking() then return false end
+
+            -- check that there are sufficient targets to AE taunt and that it is (optionally) safe to do so
+            local mobs = mq.TLO.SpawnCount("NPC radius 50 zradius 50")()
+            if (RGMercUtils.GetSetting('SafeAETaunt') and mobs > RGMercUtils.GetXTHaterCount()) or RGMercUtils.GetXTHaterCount() < RGMercUtils.GetSetting('AETauntCnt') then return false end
+
+            --check to make sure the above targets need to be tanked
+            local xtCount = mq.TLO.Me.XTarget()
+            for i = 1, xtCount do
+                local xtSpawn = mq.TLO.Me.XTarget(i)
+                --ensure we only check hostiles, the hostile is in range, that we aren't already tanking it, and that the other xtargets aren't out of range of an AE taunt
+                if xtSpawn() and (xtSpawn.ID() or 0) > 0 and (xtSpawn.TargetType() or ""):lower() == "auto hater"
+                    and (xtSpawn.Distance() or 999) <= 50 and xtSpawn.PctAggro() < 100 and mobs > 1 then
+                    return true

```

Comment:
check mobs > 1 before all to to avoid all that processing if we don't even have more than 1 mob. in fact that should just early out I think

Reply:
Added in 7d7ac2957123d84f244badbff7dcd1bf42aabaf2